dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/document/visibility/ArgumentChecker.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.rya.mongodb.document.visibility;
+
+/**
+ * This class provides methods to check arguments of a variable number for null values, or anything else that might be required on a routine basis. These
+ * methods should be used for early failures as close to the end user as possible, so things do not fail later on the server side, when they are harder to
+ * debug.
+ *
+ * Methods are created for a specific number of arguments, due to the poor performance of array allocation for varargs methods.
+ */
+public class ArgumentChecker {
```

Comment:
This is a straight copy from accumulo-core so we wouldn't need to add that dependency for mongodb components.  All the classes in the org.apache.rya.mongodb.document.visibility package are from accumulo-core and I didn't really modify any of them beside the ColumnVisibility class.

I could add the accumulo-core dependency and we'd have less classes to maintain.